CMakeLists.txt

```diff
@@ -1,5 +1,12 @@
 cmake_minimum_required(VERSION 3.8)
 
+# use ccache if available
+find_program(CCACHE_PROGRAM ccache)
+if(CCACHE_PROGRAM)
+    message(STATUS "Found ccache in ${CCACHE_PROGRAM}")
+    set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE "${CCACHE_PROGRAM}")
+endif()
+
```

Comment:
Does this need to happen before the call to `project()`?

Reply:
I don't think so. Should it be moved below?

Reply:
Preferably so. By putting it before `project()` we're insinuating that it _must_ be there when that is likely not the case. I'd prefer to put as much code after `project()` as possible since most code need not exist before it.

Reply:
Done

Reply:
Did you mean to move this after `project()`?

Reply:
@ChrisThrasher, sorry -- should be fine now.